Spigot-Server-Patches/0492-Added-new-BlockFailedDispenseEvent.patch

```diff
@@ -0,0 +1,32 @@
+From 9403d0fc108fd3be4a60ff24e7ed16a0e057e5b6 Mon Sep 17 00:00:00 2001
+From: TheViperShow <29604693+TheViperShow@users.noreply.github.com>
+Date: Tue, 21 Apr 2020 23:00:58 +0200
+Subject: [PATCH] Added new BlockFailedDispenseEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/IDispenseBehavior.java b/src/main/java/net/minecraft/server/IDispenseBehavior.java
+index 3af686c7..e6d6e00f 100644
+--- a/src/main/java/net/minecraft/server/IDispenseBehavior.java
++++ b/src/main/java/net/minecraft/server/IDispenseBehavior.java
+@@ -1,5 +1,6 @@
+ package net.minecraft.server;
+ 
++import com.destroystokyo.paper.event.block.BlockFailedDispenseEvent;
+ import java.util.Iterator;
+ import java.util.List;
+ import java.util.Random;
+@@ -16,6 +17,11 @@ import org.bukkit.event.world.StructureGrowEvent;
+ public interface IDispenseBehavior {
+ 
+     IDispenseBehavior NONE = (isourceblock, itemstack) -> {
++        // Paper start - register empty item
++        World world = isourceblock.getWorld();
++        org.bukkit.block.Block bukkitBlock = world.getWorld().getBlockAt(isourceblock.getBlockPosition().getX(), isourceblock.getBlockPosition().getY(), isourceblock.getBlockPosition().getZ());
++        BlockFailedDispenseEvent blockFailedDispenseEvent = new BlockFailedDispenseEvent(bukkitBlock);
```

Comment:
You don't do anything with the event instance. You need to call it. I think it's: `Bukkit.getPluginManager().callEvent(Event)` And if it is `Cancellable` you need to check whether it's cancelled and act based on that.

Reply:
Aikar told me that I'm a stupid Bukkit guy who can't use Paper, so what you actually need to do is just `Event#callEvent`, which also returns whether the event is cancelled.